File: src/dere_plugin/skills/recalling-past-context/scripts/entity_search.py
#!/usr/bin/env python3
"""Search entity timeline."""

from __future__ import annotations

import json
import sys
from pathlib import Path

import requests

sys.path.insert(0, str(Path(__file__).parent.parent.parent / "scripts"))
from config_reader import get_daemon_url


def search_entity(entity_name: str) -> dict | None:
    """Search for entity timeline."""
    daemon_url = get_daemon_url()

    try:
        response = requests.get(
            f"{daemon_url}/entities/timeline/{entity_name}",
            timeout=5
        )
        response.raise_for_status()
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error searching entity: {e}", file=sys.stderr)
        return None


def main():
    """Main entry point."""
    if len(sys.argv) < 2:
        print("Usage: entity_search.py <entity_name>", file=sys.stderr)
        sys.exit(1)

    entity_name = sys.argv[1]
    result = search_entity(entity_name)

    if result:
        print(json.dumps(result, indent=2))
    else:
        sys.exit(1)


if __name__ == "__main__":
    main()

File: plugins/dere_core/skills/recall/scripts/entity_search.ts
import { getDaemonUrl } from "../../../scripts/config_reader.js";

async function searchEntity(entityName: string): Promise<Record<string, unknown> | null> {
  const daemonUrl = await getDaemonUrl();
  const controller = new AbortController();
  const timeout = setTimeout(() => controller.abort(), 5_000);
  try {
    const response = await fetch(`${daemonUrl}/kg/entity/${entityName}`, {
      method: "GET",
      signal: controller.signal,
    });
    if (!response.ok) {
      console.error(`Error searching entity: ${response.status} ${response.statusText}`);
      return null;
    }
    return (await response.json()) as Record<string, unknown>;
  } catch (error) {
    console.error(`Error searching entity: ${String(error)}`);
    return null;
  } finally {
    clearTimeout(timeout);
  }
}

async function main(): Promise<void> {
  const entityName = process.argv[2];
  if (!entityName) {
    console.error("Usage: entity_search.ts <entity_name>");
    process.exit(1);
  }

  const result = await searchEntity(entityName);
  if (result) {
    console.log(JSON.stringify(result, null, 2));
  } else {
    process.exit(1);
  }
}

if (import.meta.main) {
  void main();
}
